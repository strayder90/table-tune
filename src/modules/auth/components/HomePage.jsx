import {Button} from 'semantic-ui-react';
import {useNavigate} from 'react-router-dom';

const HomePage = () => {
    const navigate = useNavigate();

    const handleLogout = () => {
        localStorage.removeItem('isAuthenticated');

        navigate('/login');
    };

    return (
        <>
            <h1>Welcome to the Home Page!</h1>
            <Button color='red' onClick={handleLogout}>Log Out</Button>
        </>
    );
};

export default HomePage;
