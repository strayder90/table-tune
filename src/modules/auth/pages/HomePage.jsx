import React from 'react';
import {Button} from 'semantic-ui-react';
import {useNavigate} from 'react-router-dom';

const HomePage = () => {
    const navigate = useNavigate();

    const handleLogout = () => {
        localStorage.removeItem('isAuthenticated');

        navigate('/login');
    };

    return (
        <>
            <h1>Welcome to the Home Page!</h1>
            <Button color='red' onClick={handleLogout}>Logout</Button>
        </>
    );
};

export default HomePage;
