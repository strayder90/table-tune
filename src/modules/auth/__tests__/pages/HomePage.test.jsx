import React from 'react';
import '@testing-library/jest-dom';
import {render, screen} from '@testing-library/react';
import {MemoryRouter} from 'react-router-dom';

import HomePage from '@modules/auth/pages/HomePage.jsx';
import {userEvent} from '@testing-library/user-event';

const mockNavigate = vi.fn();
vi.doMock('react-router-dom', () => ({
    ...vi.requireActual('react-router-dom'),
    useNavigate: () => mockNavigate
}));

describe('HomePage test suite', () => {
    test('renders without crashing', () => {
        const {container} = render(
            <MemoryRouter>
                <HomePage/>
            </MemoryRouter>
        );
        expect(container).toMatchSnapshot();

        const header = screen.getByRole('heading', {name: /Welcome to the Home Page!/i});
        expect(header).toBeInTheDocument();

        const logoutButton = screen.getByRole('button', {name: /Logout/i});
        expect(logoutButton).toBeInTheDocument();
    });

    test('clicking logout button clears localStorage and redirects to login', async () => {
        localStorage.setItem('isAuthenticated', 'true');

        render(
            <MemoryRouter>
                <HomePage/>
            </MemoryRouter>
        );

        expect(localStorage.getItem('isAuthenticated')).toBe('true');

        const logoutButton = screen.getByRole('button', {name: /Logout/i});

        await userEvent.click(logoutButton);

        expect(localStorage.getItem('isAuthenticated')).toBeNull();
    });
});
