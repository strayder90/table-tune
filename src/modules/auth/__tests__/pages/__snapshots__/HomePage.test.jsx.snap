// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HomePage test suite > renders without crashing 1`] = `
<div>
  <h1>
    Welcome to the Home Page!
  </h1>
  <button
    class="ui red button"
  >
    Logout
  </button>
</div>
`;
